### DISCUSSION TITLE
Question about how combine operator works

### DISCUSSION THREAD

**Discussion by CB1915** [nextflow-io/nextflow#5317] [Q&A] [answered]
Hello all! When using the combine operator, does it wait for all iterations of a process to finish before doing the actual combining? Say I have 10 iterations going, do all 10 of those need to finish before the next process can start?

**Comment by bentsherman** [comment, marked as answer]
No, it will emit combinations as soon as they become available

**Reply by CB1915** [reply]
Thank you, I ask because it seems like work directories for the next process are not being made until the previous process has completely finished.
